src/parser/DASHTree.cpp

```diff
@@ -722,6 +723,28 @@ static void XMLCALL start(void* data, const char* el, const char** attr)
             attr += 2;
           }
 
+          if (dash->current_adaptationset_->mimeType_ == "text/vtt" &&
+              dash->current_representation_->codecs_.empty())
+            dash->current_representation_->codecs_ = "wvtt";
+
+          if (dash->current_adaptationset_->mimeType_ == "application/ttml+xml" &&
+              dash->current_representation_->codecs_.empty())
+            dash->current_representation_->codecs_ = "ttml";
+
+          if (dash->current_adaptationset_->type_ != DASHTree::SUBTITLE &&
+              dash->current_representation_->codecs_ == "wvtt")
+          {
+            dash->current_adaptationset_->type_ = DASHTree::SUBTITLE;
+            dash->current_adaptationset_->mimeType_ = "text/vtt";
+          }
+
+          if (dash->current_adaptationset_->type_ != DASHTree::SUBTITLE &&
+              dash->current_representation_->codecs_ == "ttml")
+          {
+            dash->current_adaptationset_->type_ = DASHTree::SUBTITLE;
+            dash->current_adaptationset_->mimeType_ = "application/ttml+xml";
+          }
+
```

Comment:
These could both be nested under 1 condition for if (dash->current_adaptationset_->type_ != DASHTree::SUBTITLE)